packages/flutter/test/widgets/align_test.dart

```diff
@@ -112,4 +112,70 @@ void main() {
     expect(size.width, equals(800.0));
     expect(size.height, equals(10.0));
   });
+
+  testWidgets('Align widthFactor', (WidgetTester tester) async {
+    final GlobalKey inner = GlobalKey();
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Row(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            Align(
+              widthFactor: 0.5,
+              child: Container(
+                height: 100.0,
+                width: 100.0,
+              ),
+            ),
+            Align(
+              key: inner,
+              widthFactor: 0.5,
+              child: Container(
+                height: 100.0,
+                width: 100.0,
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+    final RenderBox box = inner.currentContext.findRenderObject() as RenderBox;
+    expect(box.size.width, equals(50.0));
+  });
+
+  testWidgets('Align heightFactor', (WidgetTester tester) async {
+    final GlobalKey inner = GlobalKey();
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: <Widget>[
+            Align(
+              alignment: Alignment.center,
+              heightFactor: 0.5,
+              child: Container(
+                height: 100.0,
+                width: 100.0,
+              ),
+            ),
+            Align(
+              key: inner,
+              alignment: Alignment.center,
+              heightFactor: 0.5,
+              child: Container(
+                height: 100.0,
+                width: 100.0,
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+    final RenderBox box = inner.currentContext.findRenderObject() as RenderBox;
```

Comment:
For all, these lookups could be:
```dart
final RenderBox textBox1 = tester.renderObject<RenderBox>(find.byKey(key1));
```
Or if there were only one instance of the class:
```dart
final RenderBox box = tester.renderObject<RenderBox>(find.byType(AnimatedAlign));
```